js_modules/dagit/packages/core/src/assets/AssetView.tsx

```diff
@@ -201,94 +156,98 @@ export const AssetView: React.FC<Props> = ({assetKey}) => {
+                  requestedDepth={visible.requestedDepth}
+                  assetGraphData={visibleAssetGraph.assetGraphData}
+                  graphQueryItems={visibleAssetGraph.graphQueryItems}
+                />
+              ) : (
+                <Box style={{flex: 1}} flex={{alignItems: 'center', justifyContent: 'center'}}>
+                  <Spinner purpose="page" />
+                </Box>
+              )
+            ) : (
+              <AssetNoDefinitionState />
             )
+          ) : (params.view || defaultTab) === 'overview' ? (
+            <AssetOverview
+              assetKey={assetKey}
+              assetLastMaterializedAt={lastMaterializedAt}
+              assetHasDefinedPartitions={!!definition?.partitionDefinition}
+              params={params}
+              paramsTimeWindowOnly={!!params.asOf}
+              setParams={setParams}
+              liveData={definition ? liveDataByNode[toGraphId(definition.assetKey)] : undefined}
+            />
+          ) : params.view === 'plots' ? (
+            <AssetPlots
+              assetKey={assetKey}
+              assetHasDefinedPartitions={!!definition?.partitionDefinition}
+              params={params}
+              setParams={setParams}
+            />
           ) : (
-            <AssetNoDefinitionState />
-          )
-        ) : (params.view || defaultTab) === 'overview' ? (
-          <AssetOverview
-            assetKey={assetKey}
-            assetLastMaterializedAt={lastMaterializedAt}
-            assetHasDefinedPartitions={!!definition?.partitionDefinition}
-            params={params}
-            paramsTimeWindowOnly={!!params.asOf}
-            setParams={setParams}
-            liveData={definition ? liveDataByNode[toGraphId(definition.assetKey)] : undefined}
-          />
-        ) : params.view === 'plots' ? (
-          <AssetPlots
-            assetKey={assetKey}
-            assetHasDefinedPartitions={!!definition?.partitionDefinition}
-            params={params}
-            setParams={setParams}
-          />
-        ) : (
-          <AssetEvents
-            assetKey={assetKey}
-            assetLastMaterializedAt={lastMaterializedAt}
-            assetHasDefinedPartitions={!!definition?.partitionDefinition}
-            params={params}
-            paramsTimeWindowOnly={!!params.asOf}
-            setParams={setParams}
-            liveData={definition ? liveDataByNode[toGraphId(definition.assetKey)] : undefined}
-          />
-        ))}
+            <AssetEvents
+              assetKey={assetKey}
+              assetLastMaterializedAt={lastMaterializedAt}
+              assetHasDefinedPartitions={!!definition?.partitionDefinition}
+              params={params}
+              paramsTimeWindowOnly={!!params.asOf}
+              setParams={setParams}
+              liveData={definition ? liveDataByNode[toGraphId(definition.assetKey)] : undefined}
+            />
+          )}
+        </>
+      )}
```

Comment:
Just one opinion here, but I find these nested ternaries pretty hard to follow.

Reply:
Yeah... let me see if I can fix this before this merges 👍   I wish this used React Router sub-pages instead of a query string for each tab but that might make it difficult to feature flag...